cirq/optimizers/eject_z.py

```diff
@@ -26,6 +27,18 @@
     from typing import Dict, List, Tuple
 
 
+def _is_swaplike(op: ops.Operation):
+    gate1 = ops.op_gate_of_type(op, ops.SwapPowGate)
+    if gate1:
+        return gate1.exponent == 1
+
+    gate2 = ops.op_gate_of_type(op, ops.FSimGate)
+    if gate2:
+        return np.isclose(gate2.theta, np.pi / 2)
+
```

Comment:
Also, `ISWAP` and its inverse.

Reply:
done

Reply:
should we also capture   + np.pi * an integer ?  I don't think any of our gates restrict their parameter ranges (probably they should?)

Reply:
udpate iswap and fsim to do any multiple.

I couldn't do the same for SWAP because the unit test was reporting `cirq.SWAP(a, b)**n` was not equal to `(cirq.SWAP ** n).on(a,b)` which confused me. I just took it out for now unless you have an idea

Reply:
That sounds like a bug; please open an issue for it. Add a TODO here to do SWAP.